jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Response.java

```diff
@@ -754,7 +754,7 @@ public void addIntHeader(String name, int value)
     @Override
     public void setStatus(int sc)
     {
-        if (sc <= 0)
+        if (sc < 100 || sc > 999)
```

Comment:
I assume that if we didn't check here the core response check would eventually throw?  But probably best to check here so the throw comes at the expected method rather than a later flush